nscplugin/src/main/scala-2/scala/scalanative/nscplugin/NirGenStat.scala

```diff
@@ -157,6 +157,19 @@ trait NirGenStat[G <: nsc.Global with Singleton] { self: NirGenPhase[G] =>
         case ann if ann.symbol == LinkClass =>
           val Apply(_, Seq(Literal(Constant(name: String)))) = ann.tree
           Attr.Link(name)
+        case ann if ann.symbol == DefineClass =>
+          (ann.tree: @unchecked) match {
+            case Apply(_, Seq(Literal(Constant(name: String)))) =>
+              Attr.Define(name, None)
+            case Apply(
+                  _,
+                  Seq(
+                    Literal(Constant(name: String)),
+                    Literal(Constant(definition: String))
```

Comment:
But doesn't this all happen at the global NIR level and these are going to to happen for every compiled C file? My PR limits the settings to only the current dependency being compiled that has native code. The GC define only happens for the nativelib so they apply library by library - in the long run we need that for all the reasons I have explained in comments of one or the other PRs.

Reply:
The other thing that a delayed Descriptor can do is allow the build to inspect options and do different things based on the options which could be helpful for the example of linking to different frameworks. To encode this stuff in annotations is probably worse than in a descriptor and then you will also have to have all the logic in the build either way as that is where the logic occurs.

Reply:
Ahh, thanks, I didn't think about this. I think for now it should be sufficient to have simple macro definitions without custom values. We can revisit that feature in the future if we find a way to scope the compiler options to the relevant glue codes, like Eric's PR does. I do think that is a good idea but it would be nice if it could happen "automatically" somehow.